packages/wallet/src/services/util/connectionStatusTracker.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  BehaviorSubject,
+  Observable,
+  distinctUntilChanged,
+  fromEvent,
+  map,
+  merge,
+  of,
+  shareReplay,
+  startWith
+} from 'rxjs';
+
+export enum ConnectionStatus {
+  down = 0,
+  up
+}
+
+export type ConnectionStatusTracker = Observable<ConnectionStatus>;
+
+export interface ConnectionStatusTrackerInternals {
+  isNodeEnv?: boolean;
+  online$?: Observable<unknown>;
+  offline$?: Observable<unknown>;
+  initialStatus?: boolean;
+}
+
+/**
+ * Returns an observable that emits the online status of the browser.
+ * When running in Node, it always emits 'up'
+ *
+ * @returns {ConnectionStatusTracker} ConnectionStatusTracker
+ */
+export const createSimpleConnectionStatusTracker = ({
+  isNodeEnv = typeof window === 'undefined',
+  online$ = isNodeEnv ? of(true) : fromEvent(window, 'online'),
```

Comment:
This will emit twice in node.js due to `startWith(initialStatus)`
```suggestion
  online$ = isNodeEnv ? EMPTY : fromEvent(window, 'online'),
```

Reply:
> ce in node.js due

The current impl returns a simple BehaviorSubject for the node scenario, ignoring online$/offline$ streams.
I think this is a problem. Since we give access to the internals, we should use them in both node/browser envs.
I will make the changes